rootfs/etc/nginx/lua/balancer/chashsubset.lua

```diff
@@ -0,0 +1,92 @@
+-- Consistent hashing to a subset of nodes. Instead of returning the same node
+-- always, we return the same subset always.
+
+local resty_chash = require("resty.chash")
+local util = require("util")
+
+local _M = { name = "chashsubset" }
+
+local function build_nodeset_map(backend)
+  local nodes = {}
+  local nodeset_map = {}
+  local node_sets = {}
+  local subset_size = 3
+
+  if backend["upstream-hash-subset-size"] == 0 then
+    subset_size = tonumber(backend["upstream-hash-subset-size"])
+  end
+
+  for _, endpoint in pairs(backend.endpoints) do
+    local endpoint_string = endpoint.address .. ":" .. endpoint.port
+    table.insert(nodes, endpoint_string)
+  end
+
+  local set_count = math.ceil(#nodes/subset_size)
+  local node_count = set_count * subset_size
+
+  -- if we don't have enough nodes, we use reuse nodes in the last set to
+  -- keep the same same on all of them.
+  if #nodes < node_count then
```

Comment:
This `if` seems unnecessary. If the condition does not hold the `for` below won't do anything anyway.

Reply:
you are right :)